python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -270,21 +292,27 @@ def _build_status_cache(
     if stored_cache_value:
         # fetch the incremental new materialized partitions, and update the cached materialized
         # subset
+        new_partitions = set()
+        if (
+            last_materialization_storage_id
+            and last_materialization_storage_id > stored_cache_value.latest_storage_id
+        ):
+            new_partitions = instance.get_materialized_partitions(
+                asset_key, after_cursor=stored_cache_value.latest_storage_id
+            )
```

Comment:
now calculating this once instead of twice - once here and then once again in build_failed_and_in_progress_partition_subset

and also only calculating it if we have reason to believe there has been a new materialization since the cursor based on the asset record